lib/src/components/NxDrawer/NxDrawer.tsx

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, {
+  HTMLAttributes,
+  ReactNode,
+  useContext,
+  useEffect,
+  useRef,
+  useState
+} from 'react';
+import classnames from 'classnames';
+
+import AbstractDialog from '../AbstractDialog/AbstractDialog';
+import NxCloseButton from '../NxCloseButton/NxCloseButton';
+import withClass from '../../util/withClass';
+
+import { Props, propTypes } from './types';
+
+import './NxDrawer.scss';
+
+interface NxDrawerHeaderProps extends HTMLAttributes<HTMLElement>{
+  children: ReactNode;
+}
+
+interface NxDrawerContextValue {
+  closeDrawer: () => void;
+  open: boolean;
+}
+
+const NxDrawerContext = React.createContext<NxDrawerContextValue>({
+  closeDrawer: () => {},
+  open: false
+});
+
+const NxDrawerHeader = (props: NxDrawerHeaderProps) => {
+  const {
+    className,
+    children,
+    ...attrs
+  } = props;
+  const { closeDrawer } = useContext(NxDrawerContext);
+
+  const classes = classnames('nx-drawer-header', className);
+
+  return (
+    <header className={classes} role="none" {...attrs}>
+      <NxCloseButton className="nx-drawer-header__close-button"
+                     type="button"
+                     onClick={() => closeDrawer()}>
```

Comment:
Is there a reason this can't just be 
```suggestion
                     onClick={closeDrawer}>
```